app/src/main/java/com/framgia/quangtran/music_42/data/source/TrackDataSource.java

```diff
@@ -1,18 +1,23 @@
 package com.framgia.quangtran.music_42.data.source;
 
+import android.content.Context;
+
 import com.framgia.quangtran.music_42.data.model.Track;
 
+import java.util.List;
+
 public interface TrackDataSource {
     interface DataCallback<T> {
-        void onSuccess(T data);
+        void onSuccess(List<T> data);
 
         void onFailed(String message);
     }
 
     interface Local {
-        void getOfflineTracks(DataCallback<Track> callback);
+        void getOfflineTracks(Context context, DataCallback<Track> callback);
```

Comment:
ko nên truyền context vào param,
như vậy e se k thể viết test cho datasource được

interface cũng k nên phụ thuộc vào các lớp implement.
GetTrackLocal thì k cần thiết phải có context
```suggestion
        void getOfflineTracks(DataCallback<Track> callback);
```

Reply:
Em cần một context để cursor thao tác ạ : Cursor cursor = context.getContentResolver() .query(MediaStore.Audio.Media.EXTERNAL_CONTENT_URI...

Reply:
thay Context bằng Cursor rồi khởi tạo Curson bên View và truyền vào là được